### PR TITLE
Update README for consistency

### PR DESCRIPTION
**Pull request by nicksnyder**
`Connect Go` -> `connect-go` to be consistent with other repos.
`in connect-go` -> `with connect-go` to be consistent with previous sentence.

### CODE REVIEW

**Review by jhump — DISMISSED** [review]

**Comment by jhump** [comment]
Holding off on merging until #712 is merged so that the only commit that reports its versions is "v1.0.0-rc1" is the one that actually has that tag.

File: README.md
Comment (nicksnyder) [reply README.md:19]:
```suggestion
Service also written with connect-go.
```

File: README.md
Comment (nicksnyder) [inline README.md:16]:
```suggestion
reference server implementation of the Conformance Service written with [connect-go](https://github.com/connectrpc/connect-go).
```

**Review by jhump — APPROVED** [review]

**Pull request merged**
